### PR TITLE
This commit was primary focused on stylefixes.

### PR DESCRIPTION
**Pull request by sobolevn**
Stylefixes:
1. added `moduledocs` and class `doc-strings`
2. stylefixes
3. integrated `pylint` into the `tox`-testing process

But there are a lot of other changes:
1. added new classificators to the `setup.py`
2. added `idea/` to `.gitignore`
3. added myself to `AUTHORS`
4. changed the way `LONG_DESCRIPTION` in `setup.py` was created (file was not closed)
5. forced `# -*- coding: utf-8 -*-` in all files

### CODE REVIEW

File: multiupload/fields.py [hunk 1]
```diff
@@ -53,48 +73,81 @@ def to_python(self, data):
...
-            raise ValidationError(self.error_messages['min_num'] % {'min_num': self.min_num, 'num_files': num_files})
-        elif self.max_num and num_files > self.max_num:
-            raise ValidationError(self.error_messages['max_num'] % {'max_num': self.max_num, 'num_files': num_files})
+            raise ValidationError(
+                self.error_messages['min_num'] % {
+                    'min_num': self.min_num,
+                    'num_files': num_files,
+                }
+            )
+        elif self.max_num is not None and num_files > self.max_num:
```
Comment (Chive) [inline multiupload/fields.py]: I don't like the logic change here - it doesn't really change the logic but just makes it more complex to read.

File: multiupload/fields.py [hunk 2]
```diff
@@ -53,48 +73,81 @@ def to_python(self, data):
...
+                self.error_messages['max_num'] % {
+                    'max_num': self.max_num,
+                    'num_files': num_files,
+                }
+            )
+
         for uploaded_file in data:
-            if self.maximum_file_size and uploaded_file.size > self.maximum_file_size:
-                raise ValidationError(self.error_messages['file_size'] % {'uploaded_file_name': uploaded_file.name})
+            if self.maximum_file_size and \
```
Comment (Chive) [inline multiupload/fields.py]: please wrap the whole conditional in brackets rather than using ``\``

**Comment by Chive** [comment]
Thanks for your PR! I've added some comments, wdyt?

File: multiupload/fields.py (hunk 1 above)
Comment (sobolevn) [inline multiupload/fields.py]: Agreed.

File: multiupload/fields.py (hunk 2 above)
Comment (sobolevn) [inline multiupload/fields.py]: Sure.

**Comment by Chive** [comment]
Awesome, thank you very much!

**Pull request merged**